x-pack/plugins/transform/server/services/license.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { Logger } from 'src/core/server';
+import {
+  IKibanaResponse,
+  KibanaRequest,
+  KibanaResponseFactory,
+  RequestHandler,
+  RequestHandlerContext,
+} from 'kibana/server';
+
+import { LicensingPluginSetup } from '../../../licensing/server';
+import { LicenseType, LICENSE_CHECK_STATE } from '../../../licensing/common/types';
+
+export interface LicenseStatus {
+  isValid: boolean;
+  isSecurityEnabled: boolean;
```

Comment:
I'm not sure this is the best place to keep this setting but it was the most straight forward way I could implement it.